test/modules/charge-categories/controller.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const Lab = require('@hapi/lab');
+const { experiment, test, beforeEach, afterEach } = exports.lab = Lab.script();
+const { expect } = require('@hapi/code');
+const sandbox = require('sinon').createSandbox();
+const repos = require('../../../src/lib/connectors/repos');
+
+const chargeCategories = [{
+  billingChargeCategoryId: '3e920b4f-5563-4c7a-8583-5e1c66ab4f83',
+  reference: '2.1.1',
+  subsistenceCharge: 9700,
+  description: 'Low loss tidal abstraction of water up to and including 25,002 megalitres a year where no model applies',
+  shortDescription: 'Low loss tidal abstraction of water up to and including 25,002 megalitres a year where no model applies',
+  dateCreated: '2021-11-18T12:00:00.585Z',
+  dateUpdated: '2021-12-22T10:42:50.334Z',
+  minVolume: 0,
+  maxVolume: 25002,
+  isTidal: true,
+  lossFactor: 'Low',
+  modelTier: 'No model',
+  restrictedSource: false
+}];
+
+const controller = require('../../../src/modules/charge-categories/controller');
+
+experiment('./src/modules/change-reasons/controller.js', () => {
+  beforeEach(async () => {
+    sandbox.stub(repos.chargeCategories, 'findOneByProperties').resolves(chargeCategories);
+  });
+
+  afterEach(async () => {
+    sandbox.restore();
+  });
+
+  experiment('.getChargeCategoryByProperties', () => {
```

Comment:
I would _recommend_ adding a couple of other tests. (Not mandatory - just a suggestion)

E.g. Breach maxVolume

E.g. What happens if some values are missing from the payload.

E.g. Does it identify the **right** Charge Category given a set of inputs, from an array of available charge categories

Reply:
Those are really good points - the database is stubbed in these tests so will make it a little inaccurate to test at this level but will flag this in the ticket to test manually. 

The route defines the parameter constraints and validates these and I have added tests for this to ensure the correct payload is received. 

and the last point is probably again a manual test because the database is stubbed for these tests.